src/shared/bus-unit-util.c

```diff
@@ -1278,11 +1281,14 @@ static int bus_append_standard_input_data(sd_bus_message *m, const char *field,
         return bus_append_byte_array(m, field, decoded, sz);
 }
 
-static int bus_append_resource_limit(sd_bus_message *m, const char *field, const char *eq) {
-        const char *suffix = ASSERT_PTR(startswith(field, "Limit"));
-        int rl, r;
+static int bus_try_append_resource_limit(sd_bus_message *m, const char *field, const char *eq) {
+        int r;
+
+        const char *suffix = startswith(field, "Limit");
+        if (!suffix)
+                return 0;
```

Comment:
why the extra check here, why not do the rlimit_from_string() call immediately, and return 0 if it fails?

Reply:
This sets `suffix`. We need to check that the string starts with "Limit", so the extra step is necessary. If I'm missing something, please explain.